### PR TITLE
COCOS-169 - Add support for algo arguments

### LINKED ISSUE #169: Feature: Enable agent to receive and pass custom arguments
### Is your feature request related to a problem? Please describe.

agent doen't pass arguments currently to algo

### Describe the feature you are requesting, as well as the possible use case(s) for it.

Agent should be able to receive and pass custom arguments to the algorithm as defined in the manifest

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SammyOina**
# What type of PR is this?
This is a feature that adds support for algo arguments

## What does this do?
- **New Features**
	- Enhanced command-line interface with new flags for algorithm type, arguments, Python runtime, and requirements file.
	- Improved context management for passing algorithm arguments.
	- Added support for command-line arguments in various algorithm implementations (binary, python, wasm).

- **Bug Fixes**
	- Resolved issues related to command execution by allowing dynamic passing of arguments to algorithms.

- **Documentation**
	- Updated README and command documentation to reflect new command options and usage.

## Which issue(s) does this PR fix/relate to?

- Resolves #169 

## Have you included tests for your changes?
No
> 

## Did you document any new/modified feature?
Yes

### Notes
```mermaid
sequenceDiagram
    participant User
    participant CLI
    participant Context
    participant Algorithm

    User->>CLI: Run algorithm with options
    CLI->>Context: Store args with AlgorithmArgsToContext
    Context->>Algorithm: Initialize with args
    Algorithm->>Context: Retrieve args with AlgorithmArgsFromContext
    Algorithm->>User: Execute algorithm with provided args
```

### CODE REVIEW

File: test/manual/algo/addition.py [hunk 1]
```diff
@@ -54,13 +54,23 @@ def read_results_from_file(self, results_file):
...
+
     computation = Computation()
 
     if len(sys.argv) == 1:
         computation.compute(a, b)
         computation.save_result()
     elif len(sys.argv) == 3 and sys.argv[1] == "test":
         computation.read_results_from_file(sys.argv[2])
+    elif len(sys.argv) == 3:
+        try:
```
Comment (rodneyosodo) [inline test/manual/algo/addition.py:66]: Wrap the try-except block only on argument parsing

File: test/manual/algo/README.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ go run ./test/computations/main.go ./test/manual/algo/addition.py public.pem fal
 ```
 
 ```bash
-./build/cocos-cli algo ./test/manual/algo/addition.py ./private.pem -a python
+./build/cocos-cli algo ./test/manual/algo/addition.py ./private.pem -a python --args="100,20"
```
Comment (rodneyosodo) [inline test/manual/algo/README.md:92]: Can we modify the algo to take something like

```bash
python addition.py --a 100 --b 20
```

As it seems the best use case for using arguments

File: cli/README.md [hunk 3]
```diff
@@ -65,6 +65,12 @@ To upload an algorithm, use the following command:
 ./build/cocos-cli agent algo /path/to/algorithm <private_key_file_path>
 ```
 
+##### Flags
+- -a, --algorithm string        Algorithm type to run (default "bin")
```
Comment (rodneyosodo) [inline cli/README.md:69]: Do we need this since the cli is self documenting?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: cli/README.md (hunk 3 above)
Comment (SammyOina) [reply cli/README.md:69]: yes we have for other files, for a quick readthrough

File: cli/algorithms.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ var (
 	pythonRuntime    string
 	algoType         string
 	requirementsFile string
+	algoArgs         []string
```
Comment (rodneyosodo) [inline cli/algorithms.go:22]: I have a wasm module that can be run by

```bash
wasmtime run ./iris-inference.wasm '{"sepal_length": 7.0, "sepal_width": 3.2, "petal_length": 4.7, "petal_width": 1.4}'
```

I am unable to upload the algorithm with `'{"sepal_length": 7.0, "sepal_width": 3.2, "petal_length": 4.7, "petal_width": 1.4}'` as an argument

```bash
./build/cocos-cli algo ./iris-inference.wasm ./private.pem -a wasm --args="{\"sepal_length\": 7.0, \"sepal_width\": 3.2, \"petal_length\": 4.7, \"petal_width\": 1.4}"
Error: invalid argument "{\"sepal_length\": 7.0, \"sepal_width\": 3.2, \"petal_length\": 4.7, \"petal_width\": 1.4}" for "--args" flag: parse error on line 1, column 2: bare " in non-quoted-field
```

```bash
./build/cocos-cli algo ./iris-inference.wasm ./private.pem -a wasm --args="'{\"sepal_length\": 7.0, \"sepal_width\": 3.2, \"petal_length\": 4.7, \"petal_width\": 1.4}'"
Error: invalid argument "'{\"sepal_length\": 7.0, \"sepal_width\": 3.2, \"petal_length\": 4.7, \"petal_width\": 1.4}'" for "--args" flag: parse error on line 1, column 3: bare " in non-quoted-field
```

```bash
./build/cocos-cli algo ./iris-inference.wasm ./private.pem -a wasm --args='{"sepal_length": 7.0, "sepal_width": 3.2, "petal_length": 4.7, "petal_width": 1.4}'
Error: invalid argument "{\"sepal_length\": 7.0, \"sepal_width\": 3.2, \"petal_length\": 4.7, \"petal_width\": 1.4}" for "--args" flag: parse error on line 1, column 2: bare " in non-quoted-field
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: cli/algorithms.go (hunk 4 above)
Comment (SammyOina) [reply cli/algorithms.go:22]: is it possible to take non json args

Comment (SammyOina) [reply cli/algorithms.go:22]: alternatively we can add a json-args flag

Comment (SammyOina) [reply cli/algorithms.go:22]: something like `go run ./cmd/cli/main.go algo ./test/manual/algo/addition.py private.pem -a python --args sepal_length=7.0,sepal_width=3.2,petal_length=4.7,petal_width=1.4`

Comment (SammyOina) [reply cli/algorithms.go:22]: resolved this should be working now

**Review by rodneyosodo — APPROVED** [review]

**Pull request merged**